source/develop/release-management/features/storage/multipath-events.html.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Multipath Alerts
+category: feature
+authors: Fred Rolland
+feature_name: Multipath Alerts
+feature_modules: engine,vdsm
+feature_status: Design
+---
+
+# Multipath Alerts
+
+
+## Overview
+
+This feature adds the ability to display alerts to the user in
+the engine UI when some multipath device have faulty paths, when
+some multipath devices are non-operational because all paths became faulty,
+and when all multipath devices on a host recover.
+
+The purpose of this feature is to help users troubleshoot storage issues
+before they become critical when a multipath device is lost, and to
+make it easier to troubleshoot cases when some LUNs are not available
+because all paths became faulty.
+
+Currently, the user has no indication in oVirt of faulty paths.
+As a result only when all paths are faulty and the access to the storage
+domain is not working, the user will get an indication:
+- If the all the hosts in the Data Center are reporting a problem in the 
+storage domain. the storage domain status that will become 'Inactive'.
+- If only one host reports a problem in the storage domain, the status
+of the host will become 'Non Operational'.
+
+
+
+## Owner
+
+- Name: [Fred Rolland](https://github.com/rollandf)
+- Email: <frolland@redhat.com>
+
+
+## Detailed Description
+
+
+### How it works
+
+Vdsm will register to udev multipath events during storage initialization.
+Vdsm will also build a map of all the faulty paths. 
+For each event, Vdsm will update the list, according to the event.
+The details of the faulty paths will be part of the statistics provided by
+'Host.getStats' verb.
+
+Engine polls the hosts for the statistics every 15 seconds.
+Engine creates an audit log event according to the information provided
+in the statistics. Only if there are changes in the faulty paths report,
+an event will be created.
+
+
+### New vdsm API
+
+New property in 'HostStats': 'MultipathAlertsMap' which is a mapping of faulty paths
+indexed by the device mapper GUID.
```

Comment:
This repeat the next section content. Maybe replace with:

   Host.getStats will report now new "multipath" key. The contents is a map of multipath alerts.

Reply:
done